sechub-test/src/test/java/com/daimler/sechub/test/RoutesTest.java

```diff
@@ -0,0 +1,174 @@
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.daimler.sechub.sharedkernel.RoleConstants;
+
+import static org.junit.Assert.*;
+
+@SuppressWarnings("rawtypes")
+public class RoutesTest {
+	
+	private Map<Class, Map<String, ArrayList<String>>> classRoutesRolesMap = new HashMap<Class, Map<String, ArrayList<String>>>();
+	
+	@Before
+	public void before() {
+		List<Class> allClasses = getAllClassesForAnnotationType(RestController.class);
+		
+		for (Class clazz : allClasses) {
+			getClassUserRolePaths(clazz);
+		}		
+	}
+	
+	@Test
+	public void routes_assigned_to_right_roles() {
+			
+		for (Entry<Class, Map<String, ArrayList<String>>> entry : classRoutesRolesMap.entrySet()) {
+			
+			String clazzName = entry.getKey().getName();
+			Map<String, ArrayList<String>> routesEntries = entry.getValue();
+			
+			for (Entry<String, ArrayList<String>> routesEntry : routesEntries.entrySet()) {
+				String route = routesEntry.getKey();
+				String[] roles = routesEntry.getValue().toArray(new String[routesEntry.getValue().size()]);
+				
+				if (!(route.contains("anonymous") || route.contains("actuator"))) {
+					assertTrue(clazzName + " with route " + route + " should contain roles.", roles.length > 0);
+				}
+				
+				if (route.contains("anonymous") || route.contains("actuator")) {
+					assertTrue(clazzName + " with route " + route + " should not contain roles.", roles.length == 0);
+				}
+				
+				if (route.contains("/api/admin")) {
+					assertTrue(clazzName + " with route " + route + " should only contain one role, ROLE_SUPERADMIN.", roles.length == 1);
+					assertTrue(clazzName + " with route " + route + " should contain ROLE_SUPERADMIN.", Arrays.asList(roles).contains(RoleConstants.ROLE_SUPERADMIN));
+				}
+				
+				if (route.contains("/api/project") || route.contains("/api/job")) {
+					assertTrue(clazzName + " with route " + route + " should contain ROLE_USER.", Arrays.asList(roles).contains(RoleConstants.ROLE_USER));
+					assertTrue(clazzName + " with route " + route + " should contain ROLE_SUPERADMIN.", Arrays.asList(roles).contains(RoleConstants.ROLE_SUPERADMIN));
+				}
+			}	
+		}
+	}
+	
+	private List<Class> getAllClassesForAnnotationType(Class clazz) {
+		ClassPathScanningCandidateComponentProvider scanner = new ClassPathScanningCandidateComponentProvider(false);
+		scanner.addIncludeFilter(new AnnotationTypeFilter(clazz));
+		
+		Set<BeanDefinition> allRestControllerBeans = scanner.findCandidateComponents("com.daimler.sechub");
+		
+		return allRestControllerBeans.stream()
+				.map( bean -> {
+					try {
+						return Class.forName(bean.getBeanClassName());
+					} catch (Exception e) {
+						e.printStackTrace();
+					}
+					return null;
```

Comment:
I am not sure if it is okay to just return null - this would lead to null entries inside list result.
hmm.. I would appreciate a runtime exception here - maybe illegal state.